### ISSUE TITLE
b.imageMode() is sometimes ignored

### ISSUE THREAD

**Issue by trych** [basiljs/basil.js#118]
While in class for most people (~80 % maybe) `b.imageMode()` did not work. Also images were drawn from center by default, which they should not. `b.imageMode(b.CORNER)` could not fix this behavior, so it had to be manually fixed by using `b.transformImage()` to the desired position.

This *might* be connected to the InDesign version, on my CS6 at home it works fine, at the uni's CC2015 it had the issue. Unfortunately I cannot test the CC2015 really well, as I don't have it at home.

Maybe someone could give this a try in a version newer than CS6:

```js
#includepath "~/Documents/;%USERPROFILE%Documents";
#include "basiljs/bundle/basil.js";

function draw() {

  b.image('image-example.jpg', 0, 0);

}

b.go();
```

This should draw from the top left corner, but in class for most people placed the image's center at the top left corner.

**Comment by ffd8** [comment]
Issue in CC 2014 too – hmm must have always placed images into a drawn shapes.

`b.image('image-example.jpg', 0, 0);` 
Draws center out, ignoring b.imageMode(b.CORNER)

`b.image('image-example.jpg', 0, 0, 300,300); `
Draws corner out while filling proportionally
Respects both b.imageMode() changes to b.CENTER / b.CORNER

**Comment by trych** [comment]
Ok, so the problem is only happening, if the last two arguments are omitted? Do I understand you correctly?

**Comment by ffd8** [comment]
Exactly. Seems to only follow the b.imageMode() if those two arguments are present...

**Comment by trych** [comment]
Ok, now we only would need someone to check if the first version of this bug to appear is the original (pre-2014) InDesign CC or InDesign CC 2014. My guess would be InDesign CC. Could somebody with a newer InDesign try the same script with the InDesign CC object model?

Should be possible by setting `app.scriptPreferences.version = 9;` in the script.

**Comment by ffd8** [comment]
I think I found the issue.. has to do with the fitting options of only 3 args:
https://github.com/basiljs/basil.js/blob/master/basil.js#L4760

I did a quick test, building a 1x1 in the 0,0 corner. Placed an image, checked the fitting options from contextual menu.. Only useful option is 'fit frame to content' (what's being using).. and that creates a centered image, regardless of the anchorpoint prefs. 

Sooooo.. it seems that another check for 3 args (+/- if b.CORNER) needs to be called further down for transformation back to x, y.

**Comment by trych** [comment]
> Sooooo.. it seems that another check for 3 args (+/- if b.CORNER) needs to be called further down for transformation back to x, y.

And there we need to make sure it is an absolute transformation, not relative, else it will mess up for CS6.

Unfortunately for me this will be rather difficult to fix, since I only have access to CS6. 😝
